stream/address_extractor.js

```diff
@@ -29,59 +29,69 @@ var through = require('through2'),
     Document = require('pelias-model').Document,
     idOrdinal = 0; // used for addresses lacking an id (to keep them unique)
 
-function hasValidAddress( item ){
-  if( !isObject( item ) ){ return false; }
-  if( !isObject( item.address ) ){ return false; }
-  if( 'string' !== typeof item.address.number ){ return false; }
-  if( 'string' !== typeof item.address.street ){ return false; }
-  if( !item.address.number.length ){ return false; }
-  if( !item.address.street.length ){ return false; }
+function hasValidAddress( doc ){
+  if( !isObject( doc ) ){ return false; }
+  if( !isObject( doc.address ) ){ return false; }
+  if( 'string' !== typeof doc.address.number ){ return false; }
```

Comment:
It might be cleaner to rewrite the body of this function as a single conditional, like:

```javascript
return isObject(doc) && isObject(doc.address) /* etc */
```